### PR TITLE
Granular role mentions

### LINKED ISSUE #213: Feature Request: Granular Mentionable Roles
This is an addition that would support things like @Staff mentions. Currently, in order to mention a role it must be set as globally mentionable on the server - this has caused a number of incidents where people have used it mistakenly to ask questions, etc. 

This feature would, by default, make all "mentionable" roles un-mentionable within Discord - then, users or roles with the correct designation (Associate and up?) could use a command like "!mention staff" or just "!staff", which would toggle the role mention on, have the bot send a message, then toggle the role mention off again (the same technique is used for the TCD announce bot here: https://github.com/TheCodingDen/tcdannounce).

This lets us have more granular control over who can actually mention these roles, while still making it easy for those who can.

### PR DESCRIPTION
**Pull request by Scott-Caldwell**
Closes #213.

Commands are `!mention <role>` and `!@ <role>`.

Includes a new claim: `MentionRestrictedRole`.

image

image

### CODE REVIEW

File: Modix.Data/Models/Mentions/MentionMappingEntity.cs [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        /// Indicates whether the role is mentionable and, if so, through what method.
+        /// </summary>
+        [Required]
+        public MentionabilityType Mentionability { get; set; }
+
+        /// <summary>
+        /// The unique identifier of the minimum rank role, if any, required to mention the role.
+        /// </summary>
+        [ForeignKey(nameof(MinimumRank))]
+        public ulong? MinimumRankId { get; set; }
```
[the remark this answers is not in the document]
Comment (Scott-Caldwell) [reply Modix.Data/Models/Mentions/MentionMappingEntity.cs]: Agreed that the most likely use case is just the `@Staff` mention

**Comment by Scott-Caldwell** [comment]
Updated. Removed the granular configuration in favor of a role designation mapping.

cc: @JakenVeina

File: Modix/ClientApp/src/models/moderation/RoleDesignation.ts [hunk 2]
```diff
@@ -1,5 +1,6 @@
 export enum RoleDesignation
 {
     Rank = "Rank",
-    ModerationMute = "ModerationMute"
-}
\ No newline at end of file
+    ModerationMute = "ModerationMute",
+    ResctrictedMentionability = "ResctrictedMentionability"
```
Comment (jrusbatch) [inline Modix/ClientApp/src/models/moderation/RoleDesignation.ts]: Small spelling error here, should be "RestrictedMentionability" (I like the new word, btw).

File: Modix.Bot/Modules/MentionModule.cs [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+            {
+                await Context.Channel.SendMessageAsync(MentionUtils.MentionRole(role.Id));
+            }
+            finally
+            {
+                await mentionContinuation();
+            }
+        }
+
+        [Command("mention")]
```
Comment (jrusbatch) [inline Modix.Bot/Modules/MentionModule.cs]: Missed opportunity for `!sudo @<role>`.

**Pull request merged**